status/controller.go

```diff
@@ -2,53 +2,158 @@ package status
 
 import (
 	"context"
-	"fmt"
 
+	"github.com/awslabs/operatorpkg/event"
 	"github.com/awslabs/operatorpkg/object"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	controllerruntime "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	MetricLabelGroup           = "group"
+	MetricLabelKind            = "kind"
+	MetricLabelNamespace       = "namespace"
+	MetricLabelName            = "name"
+	MetricLabelConditionType   = "type"
+	MetricLabelConditionStatus = "status"
+	MetricNamespace            = "operator"
+	MetricSubsystem            = "status_condition"
+)
+
 type Controller struct {
-	kubeClient client.Client
-	forObject  ConditionedObject
+	kubeClient         client.Client
+	forObject          Object
+	observedConditions map[reconcile.Request]ConditionSet
 }
 
-func NewController(client client.Client, forObject ConditionedObject) *Controller {
+func NewController(client client.Client, forObject Object) *Controller {
 	return &Controller{
-		kubeClient: client,
-		forObject:  forObject,
+		kubeClient:         client,
+		forObject:          forObject,
+		observedConditions: map[reconcile.Request]ConditionSet{},
 	}
 }
 
 func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
 	return controllerruntime.NewControllerManagedBy(m).
 		For(c.forObject).
-		Named(fmt.Sprintf("metrics.%s", object.GVK(c.forObject).Kind)).
+		Named("status").
```

Comment:
I'm assuming that we are calling all the controllers by the same name since controller-runtime can differentiate these controllers based on structured logging info?

Reply:
Correct.